public/static/js/events.js

```diff
@@ -0,0 +1,118 @@
+mciModule.factory('eventsService', ['$window', 'notificationService', function($window, notificationService) {
+  var nextTs = "";
+
+  var getMoreEvents = function(scope, mciRestService, timestamp, resourceId) {
+      var successHandler = function(resp) {
+        for (var i = 0; i < resp.data.length; i ++) {
+          event = resp.data[i];
+          diff = DeepDiff(event.before, event.after);
+          event.diff = (diff) ? diff : [];
+          for (var j = 0; j < event.diff.length; j++) {
+            eventLine = event.diff[j]
+            eventLine.text = getDiffText(eventLine);
+          }
+        }
+        scope.Events = scope.Events.concat(resp.data);
+        nextTs = getNextTs(resp.headers().link);
+      }
+      var errorHandler = function(resp) {
+         notificationService.pushNotification("Error loading events: " + resp.data.error, "errorHeader");
+      }
+
+      mciRestService.getEvents(timestamp, 0, { success: successHandler, error: errorHandler }, resourceId);
+    }
+
+  var setupWindow = function(scope, mciRestService) {
```

Comment:
This function doesn't belong here since it manipulates the UI. I would move this to the individual pages (it's fine that the scrolling snippet is duplicated in case we change the layout on those pages)